test/test_helper.rb

```diff
@@ -40,18 +40,56 @@ def asset.fresh?(env); false; end
 end
 
 def precompile_assets
-  ENV['RAILS_GROUPS'] = 'assets' # required for Rails 3.2
-  Dummy::Application.load_tasks
-  Rake::Task['assets:precompile'].reenable
-  Rake::Task['assets:precompile'].invoke
+  capture_io do
+    ENV['RAILS_GROUPS'] = 'assets' # required for Rails 3.2
+    Dummy::Application.load_tasks
+    Rake::Task['assets:precompile'].reenable
+
+    if Rails::VERSION::MAJOR == 3
+      Rake::Task['assets:precompile:all'].reenable
+      Rake::Task['assets:precompile:primary'].reenable
+      Rake::Task['assets:precompile:nondigest'].reenable
+    end
+
+    Rake::Task['assets:precompile'].invoke
+  end
+
+  if Rails::VERSION::MAJOR > 3
+    # Make a new manifest since assets weren't compiled before
+    config = Rails.application.config
+    path = File.join(config.paths['public'].first, config.assets.prefix)
+    new_manifest = Sprockets::Manifest.new(Rails.application.assets, path)
+    Rails.application.assets_manifest = new_manifest
+  end
+
+  assets_directory = File.expand_path("../dummy/public/assets", __FILE__)
+  raise "Asset precompilation failed" unless Dir.exists?(assets_directory)
```

Comment:
Yes. Before I added a test for the `YamlManifestContainer` class assets were only precompiled once which worked, but as soon as the code attempted to precompile them a second time it silently failed (because `assets:precompile:all` et al were not `reenabled`) hence the problem with `clear_precompiled_assets`.

I added the raise here to be more explicit with our expectations.